Apps/WatchServer/src/main/java/rapture/server/WatchRunner.java

```diff
@@ -0,0 +1,146 @@
+        }
+        thread.start();
+    }
+
+    /**
+     * Flag worker thread to stop gracefully.
+     */
+    public void stopThread() {
+        if (thread != null) {
+            Thread runningThread = thread;
+            thread = null;
+            runningThread.interrupt();
+        }
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            WatchKey watchKey = null;
+            try {
+                watchKey = watchService.take();
+                if (watchKey != null) {
+                    for (WatchEvent<?> watchEvent : watchKey.pollEvents()) {
+                        if (watchEvent.kind() == OVERFLOW) {
+                            log.error("Overflow event count: " + watchEvent.count());
+                            continue;
+                        }
+                        // TODO: Add support to run scripts with an event.
+                        // TODO: Check filepath is correct
+                        String actionUri = getActionForEvent(this.dirPath, watchEvent.kind().name());
+                        if (!actionUri.isEmpty()) {
+                            String filepath = this.dirPath + watchEvent.context();
+                            String scheme = new RaptureURI(actionUri).getScheme().name();
+                            Map<String, String> params = new HashMap<String, String>();
+                            params.put("filetoupload", filepath);
+                            if (scheme.equals(Scheme.SCRIPT.name())) {
+                                String runScript = Kernel.getScript().runScript(ContextFactory.getKernelUser(), actionUri, params);
+                                log.info("Started script: " + runScript + " with params: " + params.toString());
+                            }
+                            if (scheme.equals(Scheme.WORKFLOW.name())) {
+                                String createWorkOrder = Kernel.getDecision().createWorkOrder(ContextFactory.getKernelUser(), actionUri,
+                                        new HashMap<String, String>());
+                                log.info("Started workorder: " + createWorkOrder + " with params: " + params.toString());
+                            }
+                        } else {
+                            log.debug("Event: " + watchEvent.kind().name() + " not set for directory: " + this.dirPath);
+                        }
+                    }
+                    watchKey.reset();
+                }
+            } catch (InterruptedException e) {
+                e = null;
+            }
+        }
+    }
+
+    /**
+     * Get the config setting for the Directory path and event type
+     */
+    private String getActionForEvent(String dir, String event) {
+        String actionUri = "";
+        for (int i = 0; i < config.size(); i++) {
+            if (config.get(i).get("event").equals(event)) {
+                actionUri = (String) config.get(i).get("action");
```

Comment:
it just returns one action (actually a uri of the script or workflow being called); I added a break;